From e5004084f86afed98e7cbeebf6a7fb1098a2bd9a Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 29 Nov 2021 11:24:11 +0200
Subject: [PATCH] [SW-0] ppboot: gaudi2: arc: set iatu init flag

This patch add support to set iatu init flag in device status
register bases on fw_cfg_skip flag.

Change-Id: If8f021ed2d79c9c7b49673ab3c8c2372d9bde7e4
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 6a27d2f..26b6468 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -31,6 +31,7 @@
 #include <efuse.h>
 #include <string.h>
 #include <preboot/fit.h>
+#include <boot_if.h>
 #include "common.h"
 #include <bitops.h>
 #include "preboot/thermal.h"
@@ -113,6 +114,10 @@
 	/* Functionality used to be done in BTL */
 	init_prstn();
 
+	/* If FW skip disabled update boot status that iatu is configured */
+	if (!hl_skip_fw_cfg())
+		cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN);
+
 	gaudi2_axi_drain_ena();
 
 	/* set reset source */
